syntax = "proto2";
package rank_recannnew_pb;
message Dataset {
  repeated MsgClass msg = 1;
}
message MsgClass{
  message UserInfoClass{
    optional string uid = 1;
    repeated string click_hist = 2;
    repeated string show_hist = 3;
    repeated string tags_hist = 4;
    repeated string baidu_labels_hist = 5;
    optional string last_click_nid = 6;
    optional string last_click_title = 7;
    repeated string cluster_codes = 8;
    repeated string title_emb = 9;
    repeated string tags = 10;
    repeated string click_filter_hist = 11;
    repeated string show_filter_hist = 12;
    optional string city = 13;
    repeated string channels = 14;
    optional ItemInfoClass curr_item_info = 1000;
    repeated bytes patch = 1001;
  }
  message ItemInfoClass{
    optional string is_click = 1;
    optional string is_show = 2;
    repeated string cluster_codes = 3;
    repeated string title_emb = 4;
    optional string nid = 5;
    optional string title = 6;
    optional int64 rec_start = 7;
    optional int64 rec_end = 8;
    optional int64 ts = 9;
    optional string content = 10;
    optional string city_code = 11;
    repeated string city_code_list = 12;
    optional string bd_airec_status = 13;
    repeated bytes patch = 1001;
    optional string qu_default_value = 1002;
  }
  repeated ItemInfoClass item_info = 1;
  optional UserInfoClass user_info = 2;
}